pkg/build/webhook/generic/generic.go

```diff
@@ -52,7 +42,7 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 		if len(body) == 0 {
 			return revision, true, nil
 		}
-		var data webHookEvent
+		var data api.GenericWebHookEvent
 		if err = json.Unmarshal(body, &data); err != nil {
```

Comment:
However we do want to allow web hooks to be empty (for ease of use), so the very strict rules api server follows don't apply.